### PR TITLE
Cmap Selector Feature Add

### PR DESCRIPTION
**Pull request by ThomasSchlitt**
- Added combo box selection tool for color maps
- color maps include:
1. Rainbow
2. Symmetric
3. Viridis (color-blind friendly)
4. Plasma 
- cmaps are reversable upon user request
- Plasma & Viridis leverage matplotlib

### CODE REVIEW

**Comment by costerwi** [comment]
These seem like excellent features. I will take a closer look soon. Thanks Thomas!

File: legendKing.py [hunk 1]
```diff
@@ -181,36 +216,146 @@ def setup_scale(vpName, maxValue, minValue, guide, reverse=None,
...
+        bot_colors = generate_interpolated_hex_table(color1_ext,color1_mid,N)
+        top_colors = generate_interpolated_hex_table(color2_mid,color2_ext,N)
+        bot_colors.append('Grey80') #central pivot point should be grey
+        bot_colors.extend( top_colors ) #merge the two lists
+        
+        hex_colors = bot_colors
+    elif colormap in ['Cbs-cool', 'Cbs-warm']:   #self-created colormap tables
+        hex_colors = generate_interpolated_hex_table(color1, color2,N)
+    elif colormap in ['Viridis','Plasma']:
+        print('ARE WER HERE')
```
Comment (ThomasSchlitt) [inline legendKing.py:306]: Line 306 print-statement should be removed.

Comment (costerwi) [reply legendKing.py:306]: You can remove things like that and push the changes up to your branch before it gets merged into the main project.

File: legendKing.py [hunk 2]
```diff
@@ -181,36 +216,146 @@ def setup_scale(vpName, maxValue, minValue, guide, reverse=None,
...
+    return list(hex_colors)
 
 
 def readSettings():
     "Read settings file or create a new settings if necessary"
     if settings:
         return  # Already read
     try:
-        with open(jsonFileName) as f:
+        with open(jsonFileName, encoding='utf8') as f:
```
Comment (costerwi) [inline legendKing.py:358]: Unfortunately, the current versions of Abaqus CAE are still using Python2.7 which does not support an `encoding` argument in `open()`. We should remove this from both lines for now.

**Pull request closed without merging**